### PR TITLE
Merge PR 1934 and add some more fixes along the same lines

### PR DESCRIPTION
**Pull request by lovettchris**
I added "--gcc" option to setup.sh which bypasses install of LLVM 5 and "./build.sh --gcc" now works properly.  Fixed GCC compile issues involving std::sinf, std::cosf, etc.  And added a version check so cmake is only installed if user doesn't already have the right version.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -33,42 +29,73 @@ if [ ! -d "./external/rpclib/rpclib-2.2.1" ]; then
...
+    # variable for build output
+    build_dir=build_gcc_debug
+    # gcc tools
+    gcc_ver=$(gcc --version 2>&1 | head -n1 | cut -d ' ' -f4 | awk '{print $NF}')
+    gcc_path=$(which cmake)
+    if [[ "$gcc_path" == "" ]] ; then
+        echo "ERROR: run setup.sh to install a good version of gcc."
+        exit 1
+    fi
+    if [[ $(python $SCRIPT_DIR/tools/version.py $gcc_ver $MIN_GCC_VERSION) ]]; then
```
Comment (madratman) [inline build.sh]: I don't think using python for just comparing versions is a good idea
why not just do it all in shell https://unix.stackexchange.com/a/285928
also, I don't agree with enforcing python >3.4 for build process, while our APIs still work in 2.7

File: setup.sh [hunk 2]
```diff
@@ -26,57 +88,50 @@ if [ "$(uname)" == "Darwin" ]; then # osx
...
+    if [[ "$cmake_path" == "" ]] ; then
+        cmake_ver=0
+    fi
 
-if [ "$(uname)" == "Darwin" ]; then
-    CMAKE="$(greadlink -f cmake_build/bin/cmake)"
-else
-    CMAKE="$(readlink -f cmake_build/bin/cmake)"
+    #download cmake - v3.10.2 is not out of box in Ubuntu 16.04
+    if [[ $(python $SCRIPT_DIR/tools/version.py $cmake_ver $MIN_CMAKE_VERSION) ]]; then
```
Comment (madratman) [inline setup.sh]: same as above

**Comment by madratman** [comment]
PR works beautifully for me in 18.04. thanks Chris. 
Let's get rid of the python script though

File: setup.sh [hunk 3]
```diff
@@ -8,15 +8,77 @@ if [[ -d "llvm-source-39" ]]; then
...
+    --gcc)
+    gccBuild=true
+    shift # past argument
+    ;;
+esac
+done
+
+if $gccBuild; then
+    # gcc tools
+    gcc_ver=$(gcc --version 2>&1 | head -n1 | cut -d ' ' -f4 | awk '{print $NF}')
```
Comment (madratman) [inline setup.sh]: too complicated, me thinks
`gcc_ver=$(gcc -dumpfullversion)` gives you the same thing ;)

File: setup.sh [hunk 4]
```diff
@@ -8,15 +8,77 @@ if [[ -d "llvm-source-39" ]]; then
...
+    gcc_path=$(which cmake)
+    if [[ "$gcc_path" == "" ]] ; then
+        gcc_ver=0
+    fi
+    if [[ $(python $SCRIPT_DIR/tools/version.py $gcc_ver $MIN_GCC_VERSION) ]]; then
```
Comment (madratman) [inline setup.sh]: replace with 
```   if [ "$(printf '%s\n' "$MIN_GCC_VERSION" "$gcc_ver" | sort -V | head -n1)" = "$MIN_GCC_VERSION" ]; then ```
and remove python script

**Review by madratman — CHANGES_REQUESTED** [review]

**Comment by madratman** [comment]
ok I just made those changes in my fork https://github.com/madratman/AirSim/commit/50f2e95599f260aa2a82ff2a5189a7bc43ee5054

doing some final checks

**Comment by madratman** [comment]
Closing this and opening PR from my fork 
Your commits should remain, as I just build on them

File: build.sh (hunk 1 above)
Comment (lovettchris) [reply build.sh]: because I hate bash shell language :-) And python is more cross platform, but whatever, your call.

File: setup.sh (hunk 3 above)
Comment (lovettchris) [reply setup.sh]: much nicer :-)

**Pull request closed without merging**